meson.build

```diff
@@ -0,0 +1,87 @@
+project('tinc', 'c',
+  version: run_command('./src/git_tag.sh', check: true).stdout().strip(),
+  license: 'GPL-2.0-only',
+  meson_version: '>=0.51',
+  default_options: [
+    'c_std=c11',
+    'warning_level=3',
+    'buildtype=debugoptimized',
+  ],
+)
+
+cc = meson.get_compiler('c')
+os_name = host_machine.system()
+
+cc_defs = ['-D_GNU_SOURCE']
+cc_flags = [cc_defs]
+ld_flags = []
+
+opt_static = get_option('static')
+if opt_static.auto()
+  static = os_name == 'windows'
+else
+  static = opt_static.enabled()
+endif
+
+if static
+  ld_flags += '-static'
+endif
+
+opt_harden = get_option('hardening')
+if opt_harden
+  cc_flags += [
+    '-D_FORTIFY_SOURCE=2',
+    '-fwrapv',
+    '-fno-strict-overflow',
+    '-Wreturn-type',
+    '-Wold-style-definition',
+    '-Wmissing-declarations',
+    '-Wmissing-prototypes',
+    '-Wstrict-prototypes',
+    '-Wredundant-decls',
+    '-Wbad-function-cast',
+    '-Wwrite-strings',
+    '-fdiagnostics-show-option',
+    '-fstrict-aliasing',
+    '-Wmissing-noreturn',
+  ]
+  ld_flags += ['-Wl,-z,relro', '-Wl,-z,now']
+  if os_name == 'windows'
+    ld_flags += ['-Wl,--dynamicbase', '-Wl,--nxcompat']
+  endif
+endif
+
+cc_flags = cc.get_supported_arguments(cc_flags)
+ld_flags = cc.get_supported_link_arguments(ld_flags)
+
+add_project_arguments(cc_flags, language: 'c')
+add_project_link_arguments(ld_flags, language: 'c')
+
+build_root = meson.build_root()
+src_root = meson.source_root()
```

Comment:
https://mesonbuild.com/Reference-manual_builtin_meson.html#mesonbuild_root

Deprecated in 0.56

> Try using [meson.current_build_dir()](https://mesonbuild.com/Reference-manual_builtin_meson.html#mesoncurrent_build_dir) or [meson.project_build_root()](https://mesonbuild.com/Reference-manual_builtin_meson.html#mesonproject_build_root). In the rare cases where the root of the main project is needed, use [meson.global_build_root()](https://mesonbuild.com/Reference-manual_builtin_meson.html#mesonglobal_build_root) that has the same behaviour but with a more explicit name.